lib/tandem/tandemSimulator.js

```diff
@@ -87,7 +87,11 @@ exports.make = function(config){
         if(currBasal.deliveryType === 'temp') {
           if (currTempBasal != null) {
             currBasal.percent = currTempBasal.percent;
-            currBasal.payload =  { duration: currTempBasal.duration };
+            if(currBasal.isAssigned('payload')) {
+              currBasal.payload['duration'] =  currTempBasal.duration;
```

Comment:
Oh I remember you asked me about the style here. One thing I _would_ change is to use dot notation: `currBasal.payload.duration` instead of `currBasal.payload['duration']`. At one point we had that option set in linting, even.